plonky2/src/plonk/config.rs

```diff
@@ -46,6 +46,17 @@ pub trait Hasher<F: RichField>: Sized + Clone + Debug + Eq + PartialEq {
         Self::hash_no_pad(&padded_input)
     }
 
+    /// Hash the slice if necessary to reduce its length to ~256 bits. If it already fits, this is a
+    /// no-op.
+    fn hash_or_noop(inputs: &[F]) -> Self::Hash {
+        if inputs.len() <= 4 {
+            let inputs_bytes = HashOut::from_partial(inputs).to_bytes();
```

Comment:
If I'm understanding it right, it seems `inputs_bytes` will always be 32 bytes, but `Hash::from_bytes` might expect a different size like 25 bytes?

Reply:
Oh good point, I'll just change this to deserializing `inputs` directly.